oktaawscli/aws_auth.py

```diff
@@ -243,11 +286,11 @@ def __create_options_from(roles):
             # role[0] is the role arn, role[2] is the account alias
             options.append("[%s]: %s : %s" % (str(index + 1).ljust(2), role[2].ljust(27), role[0]))
         return options
-
     def __find_predefined_role_from(self, roles):
         # role_tuple[0] is the role arn
         found_roles = filter(lambda role_tuple: role_tuple[0] == self.role, roles)
+        if six.PY3:
+            found_roles = list(found_roles)
         if not found_roles:
             return None
-        else:
-            return next(found_roles)
+        return found_roles[0]
```

Comment:
Doesn't this fail in python 2? The iterator generated by filter is not indexable, right? Should probably have just been
```python
found_roles = filter(lambda role_tuple: role_tuple[0] == self.role, roles)
return next(found_roles, None)
```

Reply:
My bad, changing this wasn't intentional but the result of my branch digressing. Reverting it now